packages/relay-compiler/core/RelayIR.js

```diff
@@ -158,7 +157,7 @@ export type ScalarField = {
   kind: 'ScalarField',
   metadata: ?{[key: string]: mixed},
   name: string,
-  type: GraphQLLeafType,
+  type: GraphQLOutputType,
```

Comment:
I can add this explicit type, but it doesn't seem to be adding much direct value since there isn't any code leveraging this difference since the previous type wasn't correct and flow wasn't reporting any issues